dice_ml/explainer_interfaces/dice_genetic.py

```diff
@@ -0,0 +1,279 @@
+        :param yloss_type: Metric for y-loss of the optimization function. Takes "l2_loss" or "log_loss" or "hinge_loss".
+        :param diversity_loss_type: Metric for diversity loss of the optimization function. Takes "avg_dist" or "dpp_style:inverse_dist".
+        :param feature_weights: Either "inverse_mad" or a dictionary with feature names as keys and corresponding weights as values. Default option is "inverse_mad" where the weight for a continuous feature is the inverse of the Median Absolute Devidation (MAD) of the feature's values in the training set; the weight for a categorical feature is equal to 1 by default.
+
+        :param stopping_threshold: Minimum threshold for counterfactuals target class probability.
+        :param posthoc_sparsity_param: Parameter for the post-hoc operation on continuous features to enhance sparsity.
+        :param posthoc_sparsity_algorithm: Perform either linear or binary search. Takes "linear" or "binary". Prefer binary search when a feature range is large (for instance, income varying from 10k to 1000k) and only if the features share a monotonic relationship with predicted outcome in the model.
+
+        :return: A CounterfactualExamples object to store and visualize the resulting counterfactual explanations (see diverse_counterfactuals.py).
+
+        """
+        self.check_mad_validity(feature_weights)
+        self.check_permitted_range(permitted_range)
+        self.do_param_initializations(total_CFs, algorithm, features_to_vary, yloss_type, diversity_loss_type, feature_weights, proximity_weight, diversity_weight, categorical_penalty)
+
+        query_instance, test_pred = self.find_counterfactuals(query_instance, desired_class, stopping_threshold, posthoc_sparsity_param, posthoc_sparsity_algorithm)
+        return exp.CounterfactualExamples(self.data_interface, query_instance, test_pred, self.final_cfs, self.cfs_preds, self.final_cfs_sparse, self.cfs_preds_sparse, posthoc_sparsity_param, desired_class, encoding='label')
+
+    def predict_fn(self, input_instance):
+        """prediction function"""
+        temp_preds = self.model.get_output(input_instance).numpy()
+        return np.array([preds[(self.num_output_nodes - 1):] for preds in temp_preds], dtype=np.float32)
+
+    def compute_yloss(self, cfs):
+        """Computes the first part (y-loss) of the loss function."""
+        yloss = 0.0
+        for i in range(self.total_CFs):
+            if self.yloss_type == "l2_loss":
+                temp_loss = pow((self.model.get_output(cfs[i]) - self.target_cf_class), 2)[0][0]
+
+            elif self.yloss_type == "log_loss":
+                temp_logits = math.log((abs(self.model.get_output(cfs[i]) - 0.000001))/(1 - abs(self.model.get_output(self.cfs[i]) - 0.000001)))
+                temp_loss = self.target_cf_class[0][0] * (-1) * np.log(sigmoid(temp_logits)) + (1 - self.target_cf_class[0][0]) * (-1) * np.log(1 - sigmoid(temp_logits))
```

Comment:
where is `sigmoid` defined? your tests are failing because flake8 says sigmoid is not defined.